### PR TITLE
Switch mainboard code and bootloader to run on new hardware

### PR DESCRIPTION
**Pull request by timburke**
The next-generation momo hardware uses a PIC24FJ64GA306 chip as its MPU.  This is decently compatible with the PIC24F16KA101 but has some differences.  The mainboard code and bootloader have been ported to the new hardware and this pull request tracks their status.

_It should not be merged yet_ but only after integrating the sensor_log and sleep code into branch.

This PR is to show where the code is.  Bootloading and mainboard code is now working on the new chip.

Major hw changes are:
- Flash is 64KB vs 16KB
- RAM is 8KB vs 1.5KB
- IO pin assignment to peripherals is done through "Peripheral Pin Select" so there are not fixed pins assigned to the SPI port for example.  This is not true of I2c which for some reason does still have fixed pins.
- There is no EEPROM
- There are many more IO pins

Software changes:
- bootloader updated to not use EEPROM since there is none
- SPI code is reclocked to run at full-speed since there was no reason for it to run at 1/512th speed as had been the case until now.
- ioport.h file has macros now for manipulating ports.
- Updated code to use slightly different SPI registers in new chip.
- Made pic24 library retargetable between this chip and the previous 24F16KA101.

### CODE REVIEW

**Comment by timburke** [comment]
Controller sleep code is now integrated and tested on the new chip.  Controller board sleep current draw: 13 uA.

**Comment by amcgee** [comment]
13 uA, nice!!

I think we should either:
1) tag/branch the repo before integrating this, so we have a snapshot we can go back to that will compile and run on the older chips (of which there are a few in the wild) OR
2) make the build process (so mainboard_bootloader and mainboard) create executables for each chip (like we do with the 8bit guys)

**Comment by timburke** [comment]
I have integrated the sensor api into this branch and tested momosensor on the new hardware.  It works well and it lovely!!  I created a tag pointing to the last commit on dev so that we can always go back to the old board code if we need.

The PIC library code is multi-targeted.  We could update the pic24 build system to use scons and be multitargeted, which is probably a good thing to do anyway to get off of ceedling, but i'm not sure it's worth the effort right now given other priorities.

File: momo_modules/mainboard/src/main.c [hunk 1]
```diff
@@ -8,59 +8,52 @@
...
+#pragma config WPEND = WPENDMEM         // Segment Write Protection End Page Select (Write Protect from WPFP to the last page of memory)
 
-// FOSCSEL
-#pragma config FNOSC = FRC           	// Oscillator Select (8 MHz FRC oscillator (FRC)) without PLL
-#pragma config IESO  = OFF              // Internal External Switch Over bit (Internal External Switchover mode disabled (Two-Speed Start-up disabled))
+// CONFIG2
+#pragma config POSCMD = NONE            // Primary Oscillator Select (Primary Oscillator Disabled)
+#pragma config BOREN1 = EN              // BOR Override bit (BOR Enabled [When BOREN=1])
+#pragma config IOL1WAY = OFF            // IOLOCK One-Way Set Enable bit (The IOLOCK bit can be set and cleared using the unlock sequence)
+#pragma config OSCIOFNC = ON            // OSCO Pin Configuration (OSCO/CLKO/RC15 functions as port I/O (RC15))
```
Comment (amcgee) [inline momo_modules/mainboard/src/main.c]: So apparently this doesn't work with xc16-gcc version 1.21.  I get the error:
`src/main.c:32:9: error: unknown configuration setting: 'OSCIOFNC'`

**Pull request merged**